packages/git/src/node/dugite-git.ts

```diff
@@ -550,10 +550,14 @@ export class DugiteGit implements Git {
         return (await getTextContents(repositoryPath, commitish, path, { exec, env })).toString();
     }
 
-    async remote(repository: Repository): Promise<string[]> {
+    async remote(repository: Repository): Promise<string[]>;
+    async remote(repository: Repository, options: { verbose: true }): Promise<Remote[]>;
+    async remote(repository: Repository, options?: Git.Options.Remote): Promise<string[] | Remote[]> {
         await this.ready.promise;
         const repositoryPath = this.getFsPath(repository);
-        return this.getRemotes(repositoryPath);
+        const remotes = await this.getRemotes(repositoryPath);
+        const names = remotes.map(a => a.name);
+        return (options) ? remotes : names;
```

Comment:
This seems to be incorrect. We need the full `Remote` objects when `options` is defined AND `options.verbose === true`.